amaranth_boards/alchitry_pt_v2.py

```diff
@@ -0,0 +1,399 @@
+        )
+    bridge_bin = os.environ.get(
+        "ALCHITRY_BRIDGE_BIN",
+        os.path.join(os.path.dirname(loader_prgm), "pt_v2_loader.bin"),
+    )
+    return (loader_prgm, bridge_bin)
+
+
+# Based on pinouts published by Alchitry Labs, latest commit 7a6bd06d133ee2033c34f2e89a751889329da12b
+# https://github.com/alchitry/Alchitry-Labs-V2/blob/master/src/main/kotlin/com/alchitry/labs2/hardware/pinout/PtV2TopPin.kt
+# https://github.com/alchitry/Alchitry-Labs-V2/blob/master/src/main/kotlin/com/alchitry/labs2/hardware/pinout/PtV2AlphaTopPin.kt
+# https://github.com/alchitry/Alchitry-Labs-V2/blob/master/src/main/kotlin/com/alchitry/labs2/hardware/pinout/PtV2BottomPin.kt
+
+
+class AlchitryPtV2Platform(XilinxPlatform):
+    device = "xc7a100t"
+    package = "fgg484"
+    speed = "2"
+    default_clk = "clk100"
+
+    resources = [
+        Resource(
+            "clk100",
+            0,
+            Pins("W19", dir="i"),
+            Clock(100e6),
+            Attrs(IOSTANDARD="LVCMOS33"),
+        ),
+        Resource("rst_n", 0, Pins("N15", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 0, Pins("P19", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 1, Pins("P20", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 2, Pins("T21", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 3, Pins("R19", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 4, Pins("V22", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 5, Pins("U21", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 6, Pins("T20", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("led", 7, Pins("W20", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        # FTDI USB Interface
+        Resource(
+            "usb",
+            0,
+            Subsignal("rx", Pins("AA20", dir="i")),
+            Subsignal("tx", Pins("AA21", dir="o")),
+            # Note: The following D pins have different banks
+            Subsignal("d", Pins("W21 W22 Y21 Y22 F15 U7", dir="io")),  # D2-D7
+            Subsignal("rxf_n", Pins("F21", dir="o")),  # RXF#
+            Subsignal("txe_n", Pins("T3", dir="o")),  # TXE#
+            Subsignal("rd_n", Pins("T16", dir="o")),  # RD#
+            Subsignal("wr_n", Pins("U16", dir="o")),  # WR#
+            Subsignal("siwua", Pins("Y17", dir="o")),  # SIWUA
+            Attrs(IOSTANDARD="LVCMOS33"),
+        ),
+        # QSPI Flash
+        Resource(
+            "spi_flash",
+            0,
+            Subsignal("cs_n", Pins("T19", dir="o")),
+            # CLK is on a dedicated clock pin, but used as a standard IO for SPI
+            Subsignal("clk", Pins("L12", dir="o")),
+            Subsignal("dq", Pins("P22 R22 P21 R21", dir="io")),  # D0, D1, D2, D3
+            Attrs(IOSTANDARD="LVCMOS33"),
+        ),
+        # Analog differential pair (e.g., for XADC)
+        Resource("vp_vn", 0, DiffPairs("L10", "M9", dir="i")),
```

Comment:
I'm reasonably sure this can't be used as a digital resource, but I've never used XADC inputs.